flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -122,10 +124,7 @@ public RemoteInputChannel(
 	 * after this input channel is created.
 	 */
 	void assignExclusiveSegments() throws IOException {
-		checkState(initialCredit == 0, "Bug in input channel setup logic: exclusive buffers have " +
```

Comment:
It is better to supplement the similar check to avoid this action is called multiple times in practice.
Maybe we can check the available exclusive buffers should be 0 in `BufferManager` instead?